### ISSUE TITLE
Poco::Net::HTTPServerResponse

### ISSUE THREAD

**Issue by micheleselea** [pocoproject/poco#1648]
I found a "strange" behavior in HTTPServerResponse. The problem is related to the "Content-Length" header: if you `response.sendBuffer("test",strlen("test"));` the function set header with Content-Length to 4 and body test.
If you do just `response.send();` the function will send just header with no body, but if you do not set before `response.setContentLength(0);` you will obtain an HTTP response with no Content-Length header. This can slow down some HTTP client that will wait for a body that will never send,
So if you keep this in mind, it can be a function mode acceptable even if we can set in the library `setContentLength(0)` if is not set just to avoid slow down.
So after discovery this I introduce `response.setContentLength(0);` always before a `response.send();`
the point is that some cone in HTTPserver like when you return 0 in `createRequestHandler()` function do not set content-length. For example have a look to `HTTPServerConnection::sendErrorResponse`
Just to inform you about possibility to speed up HTTP connection without body

**Comment by obiltschnig** [comment]
That's intended behavior. HTTP allows you to omit the `Content-Length` header if you do not know the actual length upfront. However, in this case, you'll have to close the connection after sending all data. In case of HTTP/1.1, you'll have to disable persistent connections to close the connection after sending the body, as it is done in `sendErrorResponse()`. Normally, it's advisable to always either set the content length or enable chunked transfer encoding.

**Comment by micheleselea** [comment]
Thanks obiltschnig, in my HTTP server I usually have chunked transfer or sendBuffer, and where I have only send() I set content-Length to 0: I found the same in the websocket example, that kept me think about the content-length. So you told me in the reply that the sendErrorResponse setting keep alive to false do the job to close connection and so we do not have the client wait for body: ok I didn't think to this. So all works as expected.
Probably my problem was about a send() with a keepalive(true) and no Content-Length
